coderd/database/queries/workspaces.sql

```diff
@@ -128,63 +129,43 @@ WHERE
 	AND CASE
 		WHEN @status :: text != '' THEN
 			CASE
-				WHEN @status = 'pending' THEN
-					latest_build.started_at IS NULL
+			    -- Some workspace specific status refer to the transition
+			    -- type. By default, the standard provisioner job status
+			    -- search strings are supported.
+			    -- 'running' states
 				WHEN @status = 'starting' THEN
-					latest_build.started_at IS NOT NULL AND
-					latest_build.canceled_at IS NULL AND
-					latest_build.completed_at IS NULL AND
+				    latest_build.job_status = 'running' AND
+					-- TODO: Should we drop the interval? Hung workspaces
+					-- should still be their last state?
 					latest_build.updated_at - INTERVAL '30 seconds' < NOW() AND
 					latest_build.transition = 'start'::workspace_transition
-
-				WHEN @status = 'running' THEN
-					latest_build.completed_at IS NOT NULL AND
-					latest_build.canceled_at IS NULL AND
-					latest_build.error IS NULL AND
-					latest_build.transition = 'start'::workspace_transition
-
 				WHEN @status = 'stopping' THEN
-					latest_build.started_at IS NOT NULL AND
-					latest_build.canceled_at IS NULL AND
-					latest_build.completed_at IS NULL AND
+					latest_build.job_status = 'running' AND
+					-- TODO: Should we drop the interval? Hung workspaces
+					-- should still be their last state?
 					latest_build.updated_at - INTERVAL '30 seconds' < NOW() AND
 					latest_build.transition = 'stop'::workspace_transition
+				WHEN @status = 'deleting' THEN
+					latest_build.job_status = 'running' AND
+					-- TODO: Should we drop the interval? Hung workspaces
+					-- should still be their last state?
+					latest_build.updated_at - INTERVAL '30 seconds' < NOW() AND
+					latest_build.transition = 'delete'::workspace_transition
```

Comment:
I think these intervals are to ignore "hung jobs", but it feels odd to include that in some of the statuses, and not others.

I am going to drop the intervals, and defer to somewhere else to detect "hung" state. Like the UI or something. The `updated_at` timestamp is returned to the caller to decide.